src/xunit.performance.execution/AllocatedBytesForCurrentThread.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Reflection;
+
+namespace Microsoft.Xunit.Performance.Execution
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    internal static class AllocatedBytesForCurrentThread
+    {
+        /// <summary>
+        /// Loads the GetAllocatedBytesForCurrentThread method.
+        /// </summary>
+        /// <remarks>
+        /// Calling `gcGetAllocMethod.Invoke(null, null);` will cause an object
+        /// to be returned (it allocates a minimum object), so calling the
+        /// method back to back and computing the difference between them will
+        /// not be equals to zero (it will differ by 24 bytes).
+        /// </remarks>
+        static AllocatedBytesForCurrentThread()
+        {
+            var typeInfo = typeof(GC).GetTypeInfo();
+            var method = typeInfo.GetMethod("GetAllocatedBytesForCurrentThread",
+                            BindingFlags.Public | BindingFlags.Static)
+                      ?? typeInfo.GetMethod("GetAllocatedBytesForCurrentThread",
+                            BindingFlags.NonPublic | BindingFlags.Static);
+
+            // TODO: Add meaningful error message.
```

Comment:
I think we should at least output a `warning`, unable to capture this metric on netcoreapp with version less than 1.1. What do you think?